rigs/heft-web-rig/shared/webpack-base.config.js

```diff
@@ -0,0 +1,261 @@
+          exclude: /node_modules/,
+          use: [
+            // "For production builds it's recommended to extract the CSS from your bundle being able to
+            // use parallel loading of CSS/JS resources later on. This can be achieved by using the
+            // mini-css-extract-plugin, because it creates separate css files."
+            //
+            // "For development mode (including webpack-dev-server) you can use style-loader, because it injects
+            // CSS into the DOM using multiple <style></style> and works faster."
+            //
+            // "WARNING: Do not use style-loader and mini-css-extract-plugin together."
+            production
+              ? {
+                  loader: MiniCssExtractPlugin.loader
+                }
+              : {
+                  // Generates JavaScript code that injects CSS styles into the DOM at runtime.
+                  // The default configuration creates <style> elements from JS strings
+                  // https://www.npmjs.com/package/style-loader
+                  loader: require.resolve('style-loader')
+                },
+
+            {
+              // Translates CSS into CommonJS
+              // https://www.npmjs.com/package/css-loader
+              loader: require.resolve('css-loader'),
+              options: {
+                // 0 => no loaders (default);
+                // 1 => postcss-loader;
+                // 2 => postcss-loader, sass-loader
+                importLoaders: 2,
+
+                // Enable CSS modules:  https://github.com/css-modules/css-modules
+                modules: {
+                  // The "auto" setting has a confusing design:
+                  // - "false" disables CSS modules, i.e. ":local" and ":global" selectors can't be used at all
+                  // - "true" means magically disable CSS modules if the file extension isn't like ".module.css"
+                  //   or ".module.scss"
+                  // - a lambda disables CSS modules only if the lambda returns false; the function parameter is
+                  //   the resource path
+                  // - a RegExp disables CSS modules only if the resource path does not match the RegExp
+                  //
+                  // NOTE: Counterintuitively, if you instead set "modules=true" then CSS modules are enabled
+                  //       without magic, equivalent to "auto: () => true" instead of "auto: true"
+                  //
+                  // DEFAULT: "true" (i.e. path based magic)
+                  auto: (resourcePath) => {
+                    // Enable CSS modules...
+                    const useCssModules =
+                      // ...UNLESS the filename opts out using a file extension like "filename.global.scss"
+                      !/\.global\.\w+$/i.test(resourcePath) &&
+                      // ...UNLESS this is a .css file.
+                      !/\.css$/i.test(resourcePath);
+                    return useCssModules;
```

Comment:
This is weird enough, maybe:
```suggestion
                    // Files with ".global." in the name should NOT use modules
                    const isGlobalFileByName = /\.global\.\w+$/i.test(resourcePath);
                    // Files with the ".css" extension should NOT use modules
                    const isCssFile = /\.css$/i.test(resourcePath);
                    // Use modules (i.e.return `true`) if NEITHER of the above exceptions applies
                    const useCssModules = !isGlobalFileByName && !isCssFile;
                    return useCssModules;
```

Reply:
Why shouldn't you be allowed to write raw CSS modules? They're perfectly valid.

Reply:
@halfnibble had the viewpoint that if you want to make raw CSS modules, you should use the `.scss` file extension. The comments in this file explain the rationale:

```ts
// If someone wants CSS with modules, they can simply use the .sass file extension, since SASS
// is essentially a superset of CSS. (There's a small performance penalty for applying SASS to
// a plain CSS file, but the extra syntax validation probably justifies that cost.)
```

So the idea is that `.css` should only be used for actual traditional `.css` files without any fancy features.

Reply:
However I chatted with @iclanton and we've decided to enable `autoprefixer` for `.css` files.